src/eth/primitives/block_nonce.rs

```diff
@@ -0,0 +1,82 @@
+use ethereum_types::H64;
+use fake::Dummy;
+use fake::Faker;
+use sqlx::database::HasArguments;
+use sqlx::database::HasValueRef;
+use sqlx::encode::IsNull;
+use sqlx::error::BoxDynError;
+use sqlx::postgres::PgHasArrayType;
+use sqlx::Decode;
+
+use crate::gen_newtype_from;
+
+/// BlockNonce of an Ethereum account (wallet or contract).
+#[derive(Debug, Clone, Default, Eq, PartialEq, Hash, serde::Serialize, serde::Deserialize)]
+pub struct BlockNonce(H64);
```

Comment:
What is this for? 🤔

Reply:
I copied the code from bytecode so i didn't have to reimplement everything (so the comment might make. But it is one of the fields returned by eth_getBlockByNumber (it should be constant afaik.

Reply:
how is it different from the `Nonce` type?

Reply:
Just use the  `Nonce` type, they represent the same thing.

Reply:
They are not represented using the same data types though, a nonce is an U256, while the BlockNonce is a H64.

Reply:
So unlike an account nonce which is a number with 256 bits, the block nonce is a hash with 64 bits.

Reply:
oh now I got this, this Nonce is the mining algorithm nonce. Ok, we can have another type, but BlockNonce is not so good of a name. I suggest `MinerNonce` or `MiningNonce`.

But I'm thinking maybe we can have more general types to cover these kind of situations, like `Hashable` or something instead of a bunch of wrappers with the same types

Reply:
I renamed it to MinerNonce for now. But I agree that having more general types would be good, even for the other eth primitives. So instead of having `struct Gas(ethereum_types::U256)` we could have `struct Gas(stratus::U256)` and implement traits such as sqlx conversions and whatever else is specific to `U256` and not `Gas` directly on `stratus::U256`.